tests/judges/test_instruction_judge.py

```diff
@@ -0,0 +1,298 @@
+"""Unit tests for InstructionJudge implementation."""
+
+import unittest
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from server.judges.instruction_judge import InstructionJudge
+
+
+class TestInstructionJudge(TestCase):
+    """Test cases for InstructionJudge class."""
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_instruction_judge_initialization_basic(self, mock_make_judge):
+        """Test InstructionJudge initialization with basic parameters."""
+        mock_judge = Mock()
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Quality Judge', user_instructions='Check if response is helpful and accurate'
+        )
+
+        self.assertEqual(judge.name, 'Quality Judge')
+        self.assertEqual(judge.user_instructions, 'Check if response is helpful and accurate')
+        self.assertEqual(judge.system_instructions, 'Check if response is helpful and accurate')
+        self.assertEqual(judge.version, 1)
+        self.assertIsNone(judge.experiment_id)
+        self.assertIsNotNone(judge.id)
+        self.assertIsNotNone(judge.scorer_func)
+        self.assertEqual(judge.scorer_func, mock_judge)
+        
+        # Verify make_judge was called with correct parameters
+        mock_make_judge.assert_called_once()
+        call_args = mock_make_judge.call_args
+        self.assertIn('name', call_args.kwargs)
+        self.assertIn('instructions', call_args.kwargs)
+        self.assertIn('model', call_args.kwargs)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_instruction_judge_initialization_with_experiment(self, mock_make_judge):
+        """Test InstructionJudge initialization with experiment ID."""
+        mock_judge = Mock()
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Experiment Quality Judge',
+            user_instructions='Evaluate response quality',
+            experiment_id='exp456',
+        )
+
+        self.assertEqual(judge.name, 'Experiment Quality Judge')
+        self.assertEqual(judge.experiment_id, 'exp456')
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_instruction_judge_with_custom_model(self, mock_make_judge):
+        """Test InstructionJudge with custom model parameter."""
+        mock_judge = Mock()
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Custom Model Judge',
+            user_instructions='Test custom model',
+        )
+
+        mock_make_judge.assert_called_once()
+        call_args = mock_make_judge.call_args
+        self.assertEqual(call_args.kwargs['model'], 'anthropic:/claude-3-sonnet')
```

Comment:
This custom model does not link to the rest of the test code. Are we testing an implementation detail here?